lib/flex_commerce_api/json_api_client_extension/capture_surrogate_keys_middleware.rb

```diff
@@ -0,0 +1,20 @@
+# Captures surrogate keys headers and collects them for passing to the client.
+module FlexCommerceApi
+  module JsonApiClientExtension
+    class CaptureSurrogateKeysMiddleware < ::Faraday::Middleware
+      def call(environment)
+        @app.call(environment).on_complete do |env|
+          surrogate_keys = env.response_headers['external-surrogate-key'].split(' ') if env.response_headers['external-surrogate-key']
+
+          if surrogate_keys
+            if Thread.current[:shift_surrogate_keys].nil?
```

Comment:
If it is nil, that should mean "dont capture them" if it is an array, then add them to the array